src/Model/Config/Source/FilteredStores.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EasyTranslate\Connector\Model\Config\Source;
+
+use EasyTranslate\Connector\Model\Adminhtml\ProjectGetter;
+use EasyTranslate\Connector\Model\Config;
+use EasyTranslate\RestApiClient\Api\ApiException;
+use EasyTranslate\RestApiClient\Api\TeamApi;
+use Magento\Framework\Escaper;
+use Magento\Store\Model\Store;
+use Magento\Store\Model\System\Store as SystemStore;
+use Magento\Store\Ui\Component\Listing\Column\Store\Options;
+
+abstract class FilteredStores extends Options
+{
+    /**
+     * @var ProjectGetter
+     */
+    protected $projectGetter;
+
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var array
+     */
+    private $languagePairs;
+
+    public function __construct(
+        SystemStore $systemStore,
+        Escaper $escaper,
+        ProjectGetter $projectGetter,
+        Config $config
+    ) {
+        parent::__construct($systemStore, $escaper);
+        $this->projectGetter = $projectGetter;
+        $this->config        = $config;
+    }
+
+    protected function generateCurrentOptions(): void
+    {
+        $websiteCollection = $this->systemStore->getWebsiteCollection();
+        $groupCollection   = $this->systemStore->getGroupCollection();
+        $storeCollection   = $this->systemStore->getStoreCollection();
+
+        foreach ($websiteCollection as $website) {
+            $groups = [];
+            foreach ($groupCollection as $group) {
```

Comment:
Could we extract this `foreach` into a separate function?
Two of them extend the complexity a lot in my eyes

Reply:
This is an override of a core function, so that I don't want to touch it to ease comparisons during updates. We only change one line at the moment.